src/discord/commands/book.ts

```diff
@@ -0,0 +1,227 @@
+import roomModel, { Room } from '../../models/room.model';
+import sectionModel, { Section } from '../../models/section.model';
+import Timeblock from '../../models/timeblock.model';
+import { DateTime } from 'luxon';
+import { Types } from 'mongoose';
+
+interface TimeblockInformation {
+    startsAt: Date;
+    endsAt: Date;
+    availableCapacity: number;
+}
+
+interface SectionInformation extends Section {
+    _id?: Types.ObjectId;
+}
+
+function dateSuffix(day: number) {
+    if (day > 3 && day < 21) return 'th';
+    switch (day % 10) {
+        case 1:
+            return 'st';
+        case 2:
+            return 'nd';
+        case 3:
+            return 'rd';
+        default:
+            return 'th';
+    }
+}
+
+function getDateOptions(selectedDate?: string) {
+    const dateOptions = [];
+    let currentDate = DateTime.now().setZone('America/Toronto');
+
+    for (let i = 0; i < 14; i++) {
+        dateOptions.push({
+            label: `${currentDate.weekdayLong} - ${currentDate.monthLong} ${currentDate.day}${dateSuffix(currentDate.day)}`,
+            value: currentDate.toISODate(),
+            default: currentDate.toISODate() === selectedDate ? true : false,
+        });
+        currentDate = currentDate.plus({ days: 1 });
+    }
+    return dateOptions;
+}
+
+async function parseCommandOptions(interaction: CommandInteraction): Promise<string[] | undefined> {
+    //Parses the book command option parameters to return the corresponding Room's section ID
+    const roomName = interaction.options.getString('room-name');
+    const roomJson = await roomModel.findOne({ name: roomName !== null ? roomName : undefined });
+
+    if (roomJson === null) {
+        interaction.reply({ content: 'Invalid Room: Do `/rooms` to see all avaliable rooms!', ephemeral: true });
+        return undefined;
+    }
+
+    const sectionJson = await sectionModel.find({ roomId: roomJson!._id });
+    const specificSection = sectionJson.find((s) => s.name === interaction.options.getString('section-name'));
+
+    if (specificSection === undefined) {
+        interaction.reply({ content: 'Invalid Section: Do `/rooms` to see all avaliable rooms and sections!', ephemeral: true });
+        return undefined;
+    }
+
+    return [roomJson._id, specificSection._id];
+}
+
+async function searchTimeblocks(selectedDate: string, sectionInformation: SectionInformation, roomInformation: Room) {
+    //Function finds all available timeblocks for a given date
+
+    const currentDate = DateTime.now().setZone('America/Toronto');
+    let nextHour = '00';
+
+    if (currentDate.toISODate() === selectedDate) {
+        nextHour = String(currentDate.hour + 1);
+
+        if (nextHour.length === 1) {
+            nextHour = `0${nextHour}`;
+        }
+    }
+
+    const startDate = DateTime.fromISO(String(`${selectedDate}T${nextHour}:00`), { zone: 'America/Toronto' });
+    const endDate = DateTime.fromISO(String(`${selectedDate}T23:00`), { zone: 'America/Toronto' });
```

Comment:
I don't really like using string manipulation on dates here. Using Luxon native methods might be better. For example, this is probably more readable:
```ts
DateTime.fromISO(selectedDate, { zone: 'America/Toronto' }).set({ hour: 23 })
```

You wouldn't need to deal with padding a leading zero either (which for future reference you can use `padStart()` for)